components/eamxx/src/physics/mam/readfiles/tracer_reader_utils.hpp

```diff
@@ -438,14 +435,31 @@ inline std::shared_ptr<AbstractRemapper> create_horiz_remapper(
 
 inline std::shared_ptr<AtmosphereInput> create_tracer_data_reader(
     const std::shared_ptr<AbstractRemapper> &horiz_remapper,
-    const std::string &tracer_data_file) {
+    const std::string &tracer_data_file,
+    const TracerFileType file_type = NONE)
+{
   std::vector<Field> io_fields;
   for(int i = 0; i < horiz_remapper->get_num_fields(); ++i) {
     io_fields.push_back(horiz_remapper->get_src_field(i));
   }
   const auto io_grid = horiz_remapper->get_src_grid();
-  return std::make_shared<AtmosphereInput>(tracer_data_file, io_grid, io_fields,
+  if(file_type == VERT_EMISSION ){
+    // NOTE: If we are using a vertical emission nc file with altitude instead of lev,
+    // we must rename this tag.
+    // We need to perform a shallow clone of io_grid because tags are const in this object.
+    auto horiz_interp_src_grid =
+      io_grid->clone("tracer_horiz_interp_src_grid", true);
+    horiz_interp_src_grid->reset_field_tag_name(LEV, "altitude");
+    horiz_interp_src_grid->reset_field_tag_name(ILEV, "altitude_int");
+      return std::make_shared<AtmosphereInput>(tracer_data_file, horiz_interp_src_grid, io_fields,
                                            true);
+  } else{
+    // We do not need to rename tags in or clone io_grid for other types of files.
```

Comment:
@mingxuanwupnnl explained very well why to use the term "external forcings" for emissions (as they are applied like forcing in the equations). I think `NONE` here means a tracer file with no vertical dimension (e.g., surface emissions). We can rename `NONE` to `SURFACE`, but `NONE` should also be fine as it would mean a tracer file with a `NONE` vertical extent.

Reply:
We can rename the enum. Note that we are not using this enum for surface emissions. The reader for surface emissions was implemented by Balwinder. We worked in parallel on this, so we ended up with duplicate code. Because @bartgol is working on a general reader, it is better to just keep `NONE`, and I do not think we should add a reference for surface emissions. I will update the comments in this part of the code, as recommended by @mahf708.

@singhbalwinder, is this change okay: renaming `VERT_EMISSION` to `ELEVATED_EMISSION`? Or do we want to keep it as `EXT_FRC` in the EAM?

Reply:
Let's change it to `ELEVATED_EMISSIONS.` I would like to keep a reference to "emissions" to be clear that these are actually emissions.